src/Inputs/scene.inputManager.ts

```diff
@@ -804,100 +803,171 @@ export class InputManager {
+                // If dealing with a change to the delta, grab values for event init
+                const movementX = (inputIndex === PointerInput.DeltaHorizontal) ? currentState : 0;
+                const movementY = (inputIndex === PointerInput.DeltaVertical) ? currentState : 0;
+                // Get offsets from container
+                const offsetX = (inputIndex === PointerInput.DeltaHorizontal && elementToAttachTo) ? movementX! - elementToAttachTo.getBoundingClientRect().x : 0;
+                const offsetY = (inputIndex === PointerInput.DeltaVertical && elementToAttachTo) ? movementY! - elementToAttachTo.getBoundingClientRect().y : 0;
+                Object.defineProperties(evt, {
+                    pointerId: {
+                        value: (deviceType === DeviceType.Mouse ? 1 : deviceSlot)
+                    },
+                    clientX: {
+                        value: pointerX
+                    },
+                    clientY: {
+                        value: pointerY
+                    },
+                    movementX: {
+                        value: movementX
+                    },
+                    movementY: {
+                        value: movementY
+                    },
+                    offsetX: {
+                        value: offsetX
+                    },
+                    offsetY: {
+                        value: offsetY
+                    },
+                    x: {
+                        value: pointerX
+                    },
+                    y: {
+                        value: pointerY
+                    }
+                });
+
+                if (attachDown && inputIndex >= PointerInput.LeftClick && inputIndex <= PointerInput.RightClick && currentState == 1) {   // Pointer Down
+                    Object.defineProperties(evt, {
+                        type: {
+                            value: "pointerdown"
+                        },
+                        button: {
+                            value: (inputIndex - 2)
+                        }
+                    });
 
-        if (attachMove) {
-            elementToAttachTo.addEventListener(eventPrefix + "move", <any>this._onPointerMove, false);
+                    this._onPointerDown((evt as IPointerEvent));
+                }
 
-            // Wheel
-            this._wheelEventName =
-                "onwheel" in document.createElement("div")
-                    ? "wheel" // Modern browsers support "wheel"
-                    : (<any>document).onmousewheel !== undefined
-                    ? "mousewheel" // Webkit and IE support at least "mousewheel"
-                    : "DOMMouseScroll"; // let's assume that remaining browsers are older Firefox
+                if (attachUp && inputIndex >= PointerInput.LeftClick && inputIndex <= PointerInput.RightClick && currentState == 0) {   // Pointer Up
+                    Object.defineProperties(evt, {

```

Comment:
This creates GC ??? like in the previous functions wonder if there are ways to prevent it